src/scenes/game_view.cpp

```diff
@@ -0,0 +1,228 @@
+            (window_width - m_card_count * (GuiCard::width + m_card_gap) + m_card_gap) / 2;
+        for (std::size_t i = 0; i < m_cards.size(); i++) {
+            if (new_card_number == i) {
+                m_cards[i].border.x = window_width;
+            }
+            m_cards[i].target_position.x = offset + i * (GuiCard::width + m_card_gap);
+            m_cards[i].border.y = m_cards[i].target_position.y =
+                window_height - (GuiCard::height + 30);
+            m_cards[i].border.width = GuiCard::width;
+            m_cards[i].border.height = GuiCard::height;
+        }
+    }
+
+    void add_card(int window_width, int window_height, std::string_view path_to_texture = "") {
+        m_card_count++;
+        raylib::Image img;
+        try {
+            img.Load(path_to_texture.data());
+            img.Resize(GuiCard::width, GuiCard::height);
+        } catch (const raylib::RaylibException &) {
+            img = raylib::Image::Color(GuiCard::width, GuiCard::height, raylib::Color::Green());
+        }
+        m_cards.emplace_back(0, 0, raylib::Texture(img));
+        m_card_gap -= 10;
+        recalculate_card_rects(window_width, window_height, m_card_count - 1);
+    }
+
+    void remove_card(int window_width, int window_height) {
+        m_card_count--;
+        m_cards.pop_back();
+        m_card_gap += 10;
+        recalculate_card_rects(window_width, window_height);
+    }
+
+    void draw_cards(float frame_time) {
+        bool holding_card = false;
+        for (std::size_t i = 0; i < m_cards.size(); i++) {
+            bool card_is_red = false;
+            auto &card = m_cards[i];
+            if (!holding_card && card.border.CheckCollision(raylib::Mouse::GetPosition())) {
+                if (raylib::Mouse::IsButtonDown(MOUSE_BUTTON_LEFT)) {
+                    holding_card = card_is_red = true;
+                    card.border.SetPosition(card.border.GetPosition() + raylib::Mouse::GetDelta());
+                }
+            }
+            if (!holding_card) {
+                card.border.SetPosition(
+                    Vector2Lerp(card.border.GetPosition(), card.target_position, frame_time * 4)
+                );
+            }
+            if (card_is_red) {
+                card.border.DrawRounded(0.1f, 4, raylib::Color(0xFF0000FF));
+            } else {
+                card.border.DrawRounded(0.1f, 4, raylib::Color(0x004400FF));
+            }
+            card.texture.Draw(card.border.GetPosition(), raylib::Color(255, 255, 255, 200));
+            card.border.DrawRoundedLines(0.1f, 4, 3, raylib::Color::White());
+        }
+    }
+};
+
+class GameView : public Scene {
+private:
+    static constexpr const int button_width = 300;
+    static constexpr const int button_height = 40;
+    static constexpr const int board_width = 1200;
+    static constexpr const int board_height = 700;
+    static constexpr const char *background_image_path = "";
+    static constexpr const char *board_image_path = "";
```

Comment:
constexpr implies const

Reply:
not const char *, there were compiler warnings and i add const everywhere. replaced const int with int